src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -4106,25 +4106,16 @@ public void M(ref S global)
     public void M(out S s) => throw null;
 }
 ";
-            // Tracking issue: https://github.com/dotnet/roslyn/issues/22361
-
             if (languageVersion == LanguageVersion.CSharp10)
             {
-                CreateCompilationWithMscorlibAndSpan(text, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion)).VerifyDiagnostics(
-                    // (9,9): error CS8352: Cannot use variable 'local1' in this context because it may expose referenced variables outside of their declaration scope
-                    //         local1.M(out S local2);
-                    Diagnostic(ErrorCode.ERR_EscapeVariable, "local1").WithArguments("local1").WithLocation(9, 9)
-                    );
+                CreateCompilationWithMscorlibAndSpan(text, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion)).VerifyDiagnostics();
```

Comment:
it seems off to me that no error is given here in C# 10. I'm not sure why yet.

Reply:
Were you able to debug through and figure this out?

Reply:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/146967/194672824-74ff2997-d3a5-4588-905e-73de22f5a830.png">

I can't put a comment on this line cause it's not in the change but that is the problem. The `receiver` is not in the `escapedArgs` list so it's not being factored during inference. Need to add it in before you do the inference. 

In the updated code path we consider the receiver just like any other arg hence everything works out. Possible we should just do the same here and keep it simple (assuming it doesn't cause a lot of test churn)

Reply:
I think we should be able to delete the special path for the receiver, and use some kind of test against what its corresponding parameter is to decide to give the same diagnostic, and thereby avoid much test churn. I am hopeful this will be the case, anyway :)

Reply:
Thanks for debugging it!